### DISCUSSION TITLE
[Rust Sdk] Minimalistic Rust Implementation Of Model Context Protocol

### DISCUSSION THREAD

**Discussion by mohanz** [modelcontextprotocol/.github#48] [Show and tell]
### Pre-submission Checklist

- [X] This post follows the MCP community guidelines

### What would you like to share?

Repo: https://github.com/AntigmaLabs/mcp-sdk

It is a minimalistic rust implementation to kick start the process, the goal is to remain agile and easy to understand.
This implementation aims to capture the core idea of MCP while maintaining compatibility with Claude Desktop.
Many optional features are not implemented yet.

Some guidelines:
- use primitive building blocks and avoid framework if possible
- keep it simple and stupid

### Relevant Links

- Github Repo: https://github.com/AntigmaLabs/mcp-sdk

**Comment by linux-china** [comment]
I created a Rust project template for MCP CLI server too:  https://github.com/linux-china/mcp-rs-template

**Reply by mohanz** [reply]
looking good, I think it will be a very good tool to debug client behavior as well.

**Comment by dsp-ant** [comment]
That's super cool. If you don't mind, I might include it into the servers README once we update it to include more community driven efforts.

**Reply by mohanz** [reply]
my pleasure and honor :)

**Comment by liamwh** [comment]
It seems there's no active maintainers for the official [rust-sdk](https://github.com/modelcontextprotocol/rust-sdk) repo, despite there being lots of interest from contributors. How can we get additional maintainers on board?
